### PR TITLE
Extract/Export Adapters: Support `safetensors`, `pt` formats

### PR DESCRIPTION
**Pull request by jambayk**
## Describe your changes
- support saving the extracted or exported weights in `safetensors` and `pt` format in addition to `npz` format.
- new utils functions to save weights to and load weights from different file formats.
- fix test for extract_adapters_as_initializers with correct pass config
- implement `__str__` method for `StrEnumBase` so that subclasses don't have to do it. It is useful for the str representation of the strenum  to be the same as its string value. The behavior is same for `StrEnum` in python>=3.11

## Checklist before requesting a review
- [x] Add unit tests for this change.
- [x] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [x] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.
- [ ] Is this PR including examples changes? If yes, please remember to update [example documentation](https://github.com/microsoft/Olive/blob/main/docs/source/examples.md) in a follow-up PR.

### CODE REVIEW

File: olive/passes/onnx/extract_adapters.py [hunk 1]
```diff
@@ -280,6 +289,27 @@ def get_sort_key(module_name: str):
 
         return packed_weights, packings
 
+    @staticmethod
+    def save_weights(weights: Dict[str, "NDArray"], output_path: str, save_format: SaveFormat) -> Path:
+        """Save the weights to a file.
```
Comment (devang-ml) [inline olive/passes/onnx/extract_adapters.py:294]: Why not just move this also in common/utils along with load_weights ?

File: olive/passes/onnx/extract_adapters.py [hunk 2]
```diff
@@ -35,6 +36,10 @@ class ExtractAdapters(Pass):
     If make_inputs is True, the adapter weights are inputs to the model and must be provided during inference.
     """
 
+    class SaveFormat(CaseInsensitiveEnum):
+        NPZ = "npz"
+        SAFETENSORS = "safetensors"
```
Comment (devang-ml) [inline olive/passes/onnx/extract_adapters.py:41]: If we move this into common/utils.py then changes need to be made to support new format will be confined to common/utils.py

File: olive/passes/onnx/extract_adapters.py (hunk 1 above)
Comment (jambayk) [reply olive/passes/onnx/extract_adapters.py]: Done

File: olive/passes/onnx/extract_adapters.py (hunk 2 above)
Comment (jambayk) [reply olive/passes/onnx/extract_adapters.py]: Done

File: test/unit_test/common/test_save_load_weights.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    output_path = tmp_path / "weights"
+    weights_file = save_weights(weights, output_path, file_format)
+
+    assert weights_file.exists()
+    assert weights_file.suffix == expected_suffix
+
+    loaded_weights = load_weights(weights_file, framework=framework)
+    assert set(weights.keys()) == set(loaded_weights.keys())
+    for key in weights:
```
[the remark this answers is not in the document]
Comment (jambayk) [reply test/unit_test/common/test_save_load_weights.py]: Resolved

**Review by devang-ml — APPROVED** [review]

**Pull request merged**